ghcide/test/exe/Main.hs

```diff
@@ -421,9 +422,8 @@ diagnosticTests = testGroup "diagnostics"
       let contentA = T.unlines [ "module ModuleA where" ]
       _ <- createDoc "ModuleA.hs" "haskell" contentA
       expectDiagnostics [("ModuleB.hs", [])]
-  , ignoreTestBecause "Flaky #2831" $ testSessionWait "add missing module (non workspace)" $ do
-      -- need to canonicalize in Mac Os
-      tmpDir <- liftIO $ canonicalizePath =<< getTemporaryDirectory
+  , testCase "add missing module (non workspace)" $
+    withTempDir $ \tmpDir -> runInDir'' lspTestCapsNoFileWatches tmpDir "." "." [] $ do
```

Comment:
This definitely deserves a comment, it's very non-obvious why we would run this with restricted capabilities (after all, it took lots of debugging and reasoning for you to figure out that we needed it!).